extension/src/lib/portBinding.js

```diff
@@ -13,51 +13,334 @@
 /* global fluid, chrome */
 "use strict";
 
-(function ($, fluid) {
+(function (fluid) {
     var gpii = fluid.registerNamespace("gpii");
 
-    // The main component to handle settings that require DOM manipulations.
-    // It contains various subcomponents for handling various settings.
+    fluid.registerNamespace("gpii.chrome.portBinding");
+
+    gpii.chrome.portBinding.type = {
+        READ: 0,
```

Comment:
I don't think these numeric constants help too much. They will be a hazard to debugging as people try to figure out what the message type means that they see in their structure, and are an open invitation to unreadable gimmicks such as "type++" : P Please return to the use of plain strings to encode the message types

Reply:
The numbers also prevent the easy coexistence of this message bus with some other open system of typed messages going over the same port